core/src/components/datetime/test/multiple/datetime.e2e.ts

```diff
@@ -0,0 +1,159 @@
+import { expect } from '@playwright/test';
+import type { E2EPage } from '@utils/test/playwright';
+import { test } from '@utils/test/playwright';
+
+const setup = async (page: E2EPage, datetimeID: string, shouldNavigate = true) => {
+  if (shouldNavigate) await page.goto('/src/components/datetime/test/multiple/');
+  const datetime = page.locator(`#${datetimeID}`);
+  await datetime.scrollIntoViewIfNeeded();
+  await page.waitForSelector(`#${datetimeID}.datetime-ready`);
+  return datetime;
+};
+
+const screenshotDatetime = async (page: E2EPage, datetimeID: string) => {
+  const datetime = await setup(page, datetimeID);
+  expect(await datetime.screenshot()).toMatchSnapshot(
+    `datetime-multiple-${datetimeID}-${page.getSnapshotSettings()}.png`
+  );
+};
+
+test.describe('datetime: multiple date selection (visual regressions)', () => {
+  test('single default value should not have visual regressions', async ({ page }) => {
+    await screenshotDatetime(page, 'singleDefaultValue');
+  });
+
+  test('multiple default values should not have visual regressions', async ({ page }) => {
+    await screenshotDatetime(page, 'multipleDefaultValues');
+  });
+
+  test('header should not have visual regressions', async ({ page }) => {
+    await screenshotDatetime(page, 'withHeader');
+  });
+});
+
+test.describe('datetime: multiple date selection (functionality)', () => {
+  // eslint-disable-next-line no-empty-pattern
+  test.beforeEach(async ({}, testInfo) => {
+    test.skip(testInfo.project.metadata.rtl === true, 'Does not test LTR vs. RTL layout.');
+  });
+
+  test('clicking unselected days should select them', async ({ page }) => {
+    const datetime = await setup(page, 'singleDefaultValue');
+    const june1Button = datetime.locator('[data-month="6"][data-day="1"]');
+    const june2Button = datetime.locator('[data-month="6"][data-day="2"]');
+    const june3Button = datetime.locator('[data-month="6"][data-day="3"]');
+
+    await june2Button.click();
+    await expect(datetime).toHaveJSProperty('value', ['2022-06-01', '2022-06-02']);
```

Comment:
Optional: Do we need to do `page.waitForChanges()` here? Or perhaps we could wait for the `ionChange` event to be emitted and check the payload? I worry that this may be flaky if the component takes a long time to re-render on CI.